source/CollisionSet.cpp

```diff
@@ -221,6 +221,12 @@ const vector<Collision> &CollisionSet::Line(const Point &from, const Point &to,
 			const Government *iGov = it->body->GetGovernment();
 			if(it->body != target && iGov && pGov && !iGov->IsEnemy(pGov))
 				continue;
+			if(collisionType == CollisionType::SHIP)
+			{
+				Ship *ship = reinterpret_cast<Ship *>(it->body);
+				if(ship != target && ship->CanBeCarried() && ship->IsDisabled())
+					continue;
+			}
```

Comment:
This seems like the wrong place to put this in. I think the logic should be higher up, i.e. actively ignoring a collision hit that was found instead of changing the collision detection code to not find a collision.

Reply:
In that case, the `all` boolean parameter of this function would need to always be true.

Reply:
I'll open a PR to this branch to make the necessary CollisionSet changes, because I do actually agree that it's not the best to make this CollisionSet's responsibility.

Reply:
https://github.com/tibetiroka/endless-sky/pull/2